src/plugins/data/common/search/search_source/types.ts

```diff
@@ -110,7 +112,7 @@ export interface SearchSourceFields {
   /**
    * {@link IndexPatternService}
    */
-  index?: DataView;
+  index?: DataView | DataViewLazy;
```

Comment:
This affects any code that reads a data view from search source, causing typescript errors. I'm going to examine using DataViews internally with a custom field list.

Reply:
Just a quick note without having the ability to go into depth, this change would be redundant if for alerting no data view would be assigned initially , and just for the query a DataView with just the necessary fields that were fetched before would be assigned before the query is built .